Service/Model.SqlCE/SqlCeCategoryController.cs

```diff
@@ -0,0 +1,138 @@
+            {
+                return context.Categories
+                    .Where(c => c.Id == id)
+                    .ToList()
+                    .Select(c => c.ToDto())
+                    .FirstOrDefault();
+            }
+        }
+
+        /// <summary>
+        /// Fetches category list with target name. 
+        /// </summary>
+        /// <param name="name">Target name. </param>
+        /// <returns>Category list. </returns>
+        public IEnumerable<Interface.Category> SelectByName(string name)
+        {
+            using (var context = new TodoEntities())
+            {
+                return context.Categories
+                    .Where(c => c.Name == name).ToList()
+                    .Select(c => c.ToDto());
+            }
+        }
+
+        /// <summary>
+        /// Creates new category with target attributes.
+        /// </summary>
+        /// <param name="name">Short name. </param>
+        /// <param name="color">Preferable color. </param>
+        /// <param name="order">New priority. </param>
+        /// <returns>Category instance. </returns>
+        public Interface.Category Create(string name, Color color, int order)
+        {
+            using (var context = new TodoEntities())
+            {
+                var entity = new Category
+                                         {
+                                             Order = order,
+                                             Name = name,
+                                             Color = color.ToArgb()
+                                         };
+                var result = context.Categories.Add(entity);
+                context.SaveChanges();
+                return
+                    context.Categories.Where(c => c.Id == result.Id)
+                        .ToList()
+                        .Select(c => c.ToDto())
+                        .FirstOrDefault();
+            }
+        }
+
+        /// <summary>
+        /// Update category with target attributes.
+        /// </summary>
+        /// <param name="category">Update category local instance.</param>
+        public void Update(Interface.Category category)
+        {
+            using (var context = new TodoEntities())
+            {
+                context.Categories
+                    .Where(c => c.Id == category.Id)
+                    .ToList()
+                    .ForEach(c => 
+                            {
+                                c.Name = category.Name;
+                                c.Color = category.Color.ToArgb();
+                            });
+                context.SaveChanges();
+            }
+        }
+
+        /// <summary>
+        /// Delete category by primary key.
+        /// </summary>
+        /// <param name="id">Primary key of category. </param>
+        public void Delete(int id)
+        {
+            using (var context = new TodoEntities())
+            {
+                context.Categories
```

Comment:
In this case the method deletes one or zero records. 
Standard code like
```cs
var category = new Category { Id = id };
context.Categories.Attach(category);
context.Categories.Remove(category);
context.SaveChanges();
```
retrieves one record and deletes it. In case of phantom record (typically for parallely clients) this code block throws exception.
Code like
```cs
context.Categories
   .Where(c => c.Id == id)
   .ToList();
```
translated into query like
```sql
SELECT * FROM Category WHERE Id = @Id
```
and return one or zero recordes.